src/cs3iface.py

```diff
@@ -22,37 +23,48 @@
 
 # module-wide state
 ctx = {}        # "map" to store some module context: cf. init()
-tokens = {}     # map userid [string] to authentication token
+tokens = {}     # map userid [string] to {authentication token, token expiration time}
 
 
 def init(config, log):
   '''Init module-level variables'''
   ctx['log'] = log
   ctx['chunksize'] = config.getint('io', 'chunksize')
+  ctx['authtokenvalidity'] = config.getint('cs3', 'authtokenvalidity')
   revahost = config.get('cs3', 'revahost')
   # prepare the gRPC connection
   ch = grpc.insecure_channel(revahost)
   ctx['cs3stub'] = cs3gw_grpc.GatewayAPIStub(ch)
 
 
+async def _async_cleanuptokens():
+  '''A local function to go through existing tokens and remove the expired ones'''
+  now = time.time()
+  expiredkeys = [u for u in tokens if tokens[u]['exp'] < now]
+  for u in expiredkeys:
```

Comment:
Is u=userid here?

Reply:
Yes, in both places. At the same time, we could say "it doesn't matter" ;-) - that's the beauty of "mute" variables, whose scope is only a block or just a single expression (like in `[u for u in ...]`).

Reply:
Naming is (according to me) the second most interesting part of programming (architecture comes first) so I think it matters everywhere.

Reply:
Definitely agreed that naming is important :-) that's why I like mute variables or iterators to be short and concise! (cf. `for i in ...`).
Readability of `[userid for userid in tokens if tokens[userid]['exp'] < now]` is poorer IMHO.

Reply:
Coming from a verbose place I don't fully agree, but I see your point and don't have too strong opinions here 😄